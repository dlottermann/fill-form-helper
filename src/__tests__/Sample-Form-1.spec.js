const { readFileSync } = require('fs');
const { resolve } = require('path');

describe('renders html in jsdom', function() {
  let body;

  beforeEach(() => {
    const html = readFileSync(resolve(__dirname, './sample-form-1.html'), 'utf8');
    window.document.body.innerHTML = html.toString();;
    body = window.document.body;
  });

  it('should have a form with one input and a submit button', function() {
    expect(body.querySelectorAll('form')).toHaveLength(1);
    expect(body.querySelectorAll('input[type="text"]')).toHaveLength(1);
    expect(body.querySelectorAll('button[type="submit"]')).toHaveLength(1);
  });

  it('should have the form-1 sample DOM nodes', () => {
    expect(body).toMatchSnapshot();
  });
});

