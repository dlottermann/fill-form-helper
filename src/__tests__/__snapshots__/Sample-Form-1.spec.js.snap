// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders html in jsdom should have the form-1 sample DOM nodes 1`] = `
<body>
  <div
    class="sample-1"
  >
    
  
    <form>
      
    
      <input
        name="firstname"
        type="text"
      />
      
    
      <button
        type="submit"
      >
        
      Go!
    
      </button>
      
  
    </form>
    

  </div>
  


</body>
`;
